### DISCUSSION TITLE
How to use a system proxy to migrate an external git repository?

### DISCUSSION THREAD

**Discussion by proofy** [gogs/gogs#7088] [Q&A] [answered]
My gogs server have only direct access to the internet over a proxy server.
Is it possible to use these system proxy with gogs?
If yes, how to config gogs to use a system proxy?

**Comment by proofy** [comment, marked as answer]
I just found a solution.
Putting the environment vars like HTTP_PROXY into a shell script in /etc/profile.d/proxy.sh
(source https://www.serverlab.ca/tutorials/linux/administration-linux/how-to-set-environment-variables-in-linux/)
So the gogs user will use this and also the gogs server.
Be sure, that the gogs user can make DNS requests.

**Comment by xtao** [comment]
Maybe try:
- login `git` user in shell, 
- and set `git config --global https.proxy http://127.0.0.1:1235` ref: https://gist.github.com/evantoli/f8c23a37eb3558ab8765,
- then the git user will clone or pull repo with the proxy.

FYI: The environment var maybe put all http traffic through proxy.
